package conet;

message TcpServer
{
    optional string ip = 1;
    optional int32 port = 2;
    optional int32 max_conn_num=3 [default=100000];
    optional int32 enable_http=4 [default=1];
    optional int32 max_pending_num=5 [default=10000];
    optional int32 duplex=6 [default=0];
}

message UdpServer
{
    optional string ip = 1;
    optional int32 port = 2;
    optional int32 max_conn_num=3 [default=100000];
    optional int32 max_pending_num=4 [default=10000];
}

message RpcServer
{
    optional string server_name=1;
    repeated TcpServer tcp_server=2;
    repeated UdpServer udp_server=3;
    repeated TcpServer http_server=4;
    optional int32 max_pending_num=5 [default=100000];
}

message ServerGroup
{
    optional string group_name=1;
    repeated RpcServer servers=2;
    optional int32 thread_num=3 [default = 1]; // 线程数
}

message ServerConf
{
    repeated ServerGroup  server_groups = 1;
}

